### PR TITLE
Migrate CI from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by ml-evs**
Following the Travis OSS billing changes mentioned in #331, this PR migrates the Java build and checks to GH actions. It seems to work perfectly... 

We won't run out of Travis minutes for quite a while, so I've left Travis enabled until we're convinced the new Actions agree with Travis.

GitHub shows an ugly diff given the file rename, the actual changes are all in 4b14f26.

Downsides: more eggs in GH basket, but Travis is basically forcing our hand...

### CODE REVIEW

**Review by merkys — COMMENTED** [review]
I have insufficient knowledge of both Travis and GitHub Actions. Maybe someone having more experience here could review and approve?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        pip install pre-commit==2.2.0
+    - name: Run pre-commit tests
+      run: |
+        pre-commit run --all-files
```
Comment (CasperWA) [inline .github/workflows/ci.yml:28]:
```suggestion
      run: pre-commit run --all-files
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        pip install pre-commit==2.2.0
```
Comment (CasperWA) [inline .github/workflows/ci.yml:25]:
```suggestion
      run: pip install pre-commit==2.2.0
```

Comment (CasperWA) [reply .github/workflows/ci.yml:25]: Maybe we should update this version to the latest version of `2.8.2`?
```suggestion
      run: pip install pre-commit==2.8.2
```

**Review by CasperWA — APPROVED** [review]
Looks good, thanks @ml-evs !

Some slightly unrelated suggested changes. I've approved, since these suggested changes can be argued to be out-of-scope of this PR...

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    name: Test grammar and formatting
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
```
Comment (CasperWA) [inline .github/workflows/ci.yml:40]:
```suggestion
        java-version: 8
```

**Review by CasperWA — CHANGES_REQUESTED** [review]
Ah, missed this - should actually be java version 8, right? According to what is used currently in Travis?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (CasperWA) [reply .github/workflows/ci.yml:40]: Never mind, just saw [here](https://github.com/actions/setup-java#basic) that it's an alias...

**Review by CasperWA — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ml-evs) [reply .github/workflows/ci.yml:40]: Yeah, I loved that too...

**Review by shyamd — APPROVED** [review]

**Comment by ml-evs** [comment]
> Some slightly unrelated suggested changes. I've approved, since these suggested changes can be argued to be out-of-scope of this PR...

Agree with all your suggestions, but to save another round of reviewing I'll merge this as is. I'll raise another issue tracking the swagger validation of schema in the CI and can fix those minor blips then.

**Pull request merged**